lib/galaxy/objectstore/__init__.py

```diff
@@ -304,8 +312,9 @@ def size(self, obj, **kwargs):
     def delete(self, obj, entire_dir=False, **kwargs):
         path = self.get_filename(obj, **kwargs)
         extra_dir = kwargs.get('extra_dir', None)
+        obj_dir = kwargs.get('obj_dir', False)
         try:
-            if entire_dir and extra_dir:
+            if entire_dir and (extra_dir or obj_dir):
```

Comment:
I would appreciate a lot of scrutiny here, because if I messed this up, Galaxy could remove things it should not remove.